ananse/network.py

```diff
@@ -213,408 +180,117 @@ def distance_weight(self, include_promoter=False, include_enhancer=True, alpha=1
-                {
-                    "weight": [0 for z in range(remove + 1, keep1 + 1)],
-                    "dist": range(remove + 1, keep1 + 1),
-                }
-            )
-            weight3 = pd.DataFrame(
-                {
-                    "weight": [0 for z in range(1, padding - keep1 + 1)],
-                    "dist": range(keep1 + 1, padding + 1),
-                }
-            )
-
-        weight = pd.concat([weight1, weight2, weight3])
-
-        weightfile = NamedTemporaryFile(mode="w", dir=mytmpdir(), delete=False)
-        weight.to_csv(weightfile)
-        return weightfile.name
+        promoter_weight = int(include_promoter)
+        enhancer_weight = int(include_enhancer)
 
-    def aggregate_binding(self, ddf, prom, p, weight):
-
-        # ddf = dd.read_hdf(binding, key="/binding")[["factor", "enhancer", "binding"]]
-        # ddf = dd.read_csv(binding, sep="\t")[["factor", "enhancer", "binding"]]
-        prom_table = ddf.merge(prom, left_on="enhancer", right_on="loc")
-        prom_table = prom_table.groupby(["factor", "gene"])[["binding"]].max()
-        prom_table = prom_table.rename(columns={"binding": "max_binding_in_promoter"})
-        prom_table = prom_table.reset_index()
-        prom_table["source_target"] = (
-            prom_table["factor"].map(str) + "_" + prom_table["gene"].map(str)
+        weight1 = pd.DataFrame(
+            {
+                "weight": [promoter_weight for z in range(0, remove + 1)],
+                "dist": range(0, remove + 1),
+            }
         )
 
-        f_table = ddf.merge(p, left_on="enhancer", right_on="loc")
-        sum_enh = f_table.groupby(["factor", "gene"])[["binding"]].count()
-        f_table["sum_weighted_logodds"] = (
-            f_table["binding"]
-            .div(f_table["binding"].mean())
-            .apply(np.log, meta=("binding", np.float64))
-            .rmul(50000)
-            .div(f_table["dist"])
+        weight2 = pd.DataFrame(
+            {
+                "weight": [enhancer_weight for z in range(remove + 1, keep1 + 1)],
+                "dist": range(remove + 1, keep1 + 1),
+            }
         )
-        f_table["sum_logodds"] = (
-            f_table["binding"]
-            .div(f_table["binding"].mean())
-            .apply(np.log, meta=("binding", np.float64))
+
+        weight3 = pd.DataFrame(
+            {
+                "weight": [
+                    enhancer_weight
+                    * 2.0
+                    * math.exp(-u * math.fabs(z) / 1e5)
+                    / (1.0 + math.exp(-u * math.fabs(z) / 1e5))
+                    for z in range(1, padding - keep1 + 1)
+                ],
+                "dist": range(keep1 + 1, padding + 1),
+            }
         )
 
-        weight = dd.read_csv(weight)
-        f_table = f_table.merge(weight, how="left", on="dist")
-        f_table["sum_dist_weight"] = f_table["binding"] * f_table["weight"]
+        weight = pd.concat([weight1, weight2, weight3])
```

Comment:
why are this dataframes? Doesn't a single list just work? The `dist` column == index?